### PR TITLE
Added creating customers and setting owner details to Stripe even when Stripe Checkout doesn't require billing info

### PR DESCRIPTION
**Pull request by mnelson4**
## Problem this Pull Request solves

Hopefully https://github.com/eventespresso/eea-stripe-gateway/issues/4
The main change is that we send name and address info to Stripe even when the Stripe Checkout pop-up doesn't ask for billing info. 
Other changes that were needed in order to make that happen:
* Our Stripe integration now uses "sources" instead of "tokens". Sources can be reusable whereas tokens were transient. This change was needed because in order to use Stripe's API to update the customer's name and address (because the transient data wasn't updatable)
* In order to update the source's owner's name and address, we need to create customers in Stripe. We avoid creating duplicate customers based on email address (although we could just always make a new customer, Stripe doesn't mind multiple customers having the same email)
* We are now using Stripe's PHP library version 6.15 instead of 1.16 (I thought this would fix an error I was getting, but it didn't, but I figured it was a good idea to leave it in there anyway). We still modify the library's namespace so it doesn't conflict with others plugins using the name library.

## How has this been tested

* [ ] Create a paid event which does NOT require address info, and set Stripe to not require billing info. Register for the event. It should work fine. In Stripe's dashboard, view the payment. There should be a name for the "owner" (although no address info)
* [ ] Repeat the above test, using the **same email address**. It should again work and there should **not** be another customer created (we should reuse the previous one)
* [ ] Create another event which DOES require address info, but leave Stripe to NOT require billing info. Register for the event with a different email address. In the Stripe dashboard's payment, the owner name **and** address info should appear (eg https://drive.google.com/a/eventespresso.com/file/d/1w31QcI0nOwXDJlp7elya4Dt0pAdA146Z/view?usp=drivesdk)
* [ ] Now set Stripe to require billing info. Register for the first event (where address info not requested in the regform). In the Stripe dashboard payment, the owner info should again appear
* [ ] Register for the second event (which does require address info). Use different address info in the regform as compared to the Stripe pop-up billing info. In Stripe's dashboard, on the payment, you should see the owner's data is what you entered into the Stripe pop-up, not the regform

## Checklist

* [ ] I have read the documentation relating to systems affected by this pull request, see https://github.com/eventespresso/event-espresso-core/tree/master/docs
* [ ] User input is adequately validated and sanitized
* [ ] all publicly displayed strings are internationalized (usually using `esc_html__()`, see https://codex.wordpress.org/I18n_for_WordPress_Developers)
* [ ] My code is tested.
* [ ] My code follows the Event Espresso code style.
* [ ] My code has proper inline documentation.

### CODE REVIEW

File: payment_methods/Stripe_Onsite/EEG_Stripe_Onsite.gateway.php [hunk 1]
```diff
@@ -70,20 +89,62 @@ public function do_direct_payment($payment, $billing_info = null)
...
+        } else {
+            $customer = Customer::create(array(
+                "email" =>  $email,
+                "source" => $source_id,
+                'description' => $attendee->full_name()
+            ));
+            $source = $customer->sources->retrieve($source_id);
+        }
+        if (! $this->_billing_address) {
+            $source->owner->name= $attendee->full_name();
```
Comment (knazart) [inline payment_methods/Stripe_Onsite/EEG_Stripe_Onsite.gateway.php]: While at it - just missing a space here.

Comment (mnelson4) [reply payment_methods/Stripe_Onsite/EEG_Stripe_Onsite.gateway.php]: Hmm, I don't see where it's missing a space. PHPCS says its ok 🤔

**Comment by mnelson4** [comment]
re-review please!

File: payment_methods/Stripe_Onsite/EEG_Stripe_Onsite.gateway.php (hunk 1 above)
Comment (knazart) [reply payment_methods/Stripe_Onsite/EEG_Stripe_Onsite.gateway.php]: Right before the equals sign here: `$source->owner->name= $attendee->full_name();` 😄

**Comment by mnelson4** [comment]
Passing to @joshfeck for testing and feedback on UI changes. Please see https://github.com/eventespresso/eea-stripe-gateway/pull/7#issue-210025101 for the description of the changes and testing notes.

**Comment by mnelson4** [comment]
Also, a thought: we might be uncomfortable creating customers in Stripe (because it will be an unexpected and possibly unwelcome change for existing users). In that case, we could add a setting inside Stripe for that (so if a site owner chooses, we would NOT create customers in Stripe, and instead have the old behaviour).

**Comment by mnelson4** [comment]
on https://github.com/eventespresso/eea-stripe-gateway/issues/4 we realized many users will probably be alarmed at the new email going out, and this will probably add more trouble than it will solve. So closing this out.

**Pull request closed without merging**